Documentation for the Huawei Solar integration (#21352)
home-assistant/home-assistant.github.io: source/_integrations/huawei_solar.markdown

---
title: "Huawei Solar"
description: "Instructions on how to connect your Huawei SUN2000 inverter to Home Assistant"

> ha_release: "2022.3"

Reviewer: change release. hope this will be added in 2022.5
```suggestion
ha_release: "2022.5"

```

ha_category: 
    - Sensor
    - Energy
ha_iot_class: "Local Polling"
ha_config_flow: true
ha_codeowners:
  - '@wlcrs'
ha_domain: huawei_solar
---

The Huawei Solar integration connects to the Modbus interface from a Huawei SUN2000(L) inverter and integrates it into Home Assistant.

## Prerequisites

> You will need to either set a static IP on your Huawei Inverter or assign a static DHCP lease to it. Refer to 'Router connection settings' in 'Configuration' > 'Communication configuration' to set a static IP address. 

Reviewer: You should specify if dongle is needed. According to the latest firmware update MODBUS won't be supported on inverter - https://forum.huawei.com/enterprise/en/forum.php?mod=redirect&goto=findpost&ptid=789585&pid=4632813. Also take a look at https://skyboo.net/2022/02/huawei-sun2000-why-using-a-usb-dongle-for-monitoring-is-not-a-good-idea/

If understand correct your integration works only with inverter wifi or with dongle? This should be specified.
What firmware version is needed on both devices?
BTW can we use MODBUS to ETH converter without using wifi? https://www.waveshare.com/rs485-to-eth.htm or attach LAN cable directly do dongle?
FusionSolar app should work with this integration. If yes this could be mentioned.

---
title: "Huawei Solar"
description: "Instructions on how to connect your Huawei SUN2000 inverter to Home Assistant"
ha_release: "2022.3"
ha_category: 
    - Sensor
    - Energy
ha_iot_class: "Local Polling"
ha_config_flow: true
ha_codeowners:
  - '@wlcrs'
ha_domain: huawei_solar
---

> The `huawei_solar` integration connects to the Modbus-interface from a Huawei SUN2000(L) inverter and integrates it into Home Assistant.

Reviewer 2: ```suggestion
The Huawei Solar integration connects to the Modbus interface from a Huawei SUN2000(L) inverter and integrates it into Home Assistant.

```

- Shutdown Time (timestamp of the last time your inverter stopped in the evening)

If your installation has optimizers, an additional sensor 'Optimizers online' is available.

If you have a **battery** attached to your inverter, the following extra sensors are exposed (updated every 30 seconds):

- Battery Day Charge
- Battery Day Discharge
- Battery State of Capacity
- Battery Total Charge
- Battery Total Discharge
- Charge/Discharge Power
- Storage Bus Current
- Storage Bus Voltage

In case you have a Power meter installed, like the DTSU666-H, the following extra sensors are exposed (updated every 30 seconds):

- Active Power for each phase
- Total Active Power
- Grid Consumption (accumulative)
- Grid Expored (accumulative)
- Grid Frequency
- Power Factor

The following sensors are disabled by default:
- Line voltage between each of the phases
- Current for each phase
- Voltage for each phase
- Reactive Power
- Grid Reactive Power

A power meter is typically only installed when a battery is attached to the system.

## Energy dashboard

Recommended energy dashboard configuration for meter location in feed in path (`Meter location: 0`):

> - For `Grid consumption` use the `Grid Consumption` sensor from the power meter.
> - For `Return to grid` use the 'Grid Exported` sensor from the power meter.
> - For `Solar production`: 
>   - If no battery is connected to an inverter: Add each inverters `Daily Yield` sensor.
>   - If a battery is connected to an inverter: Use [Riemann sum](/integrations/integration/) over the `Input Power` sensor 
> - For `Battery systems` use the `Battery Total Charge` and `Battery Total Discharge` sensors

Reviewer 2: ```suggestion
- For "Grid consumption" use the "Grid Consumption" sensor from the power meter.
- For "Return to grid" use the "Grid Exported" sensor from the power meter.
- For "Solar production":
  - If no battery is connected to an inverter: Add each inverter's "Daily Yield" sensor.
  - If a battery is connected to an inverter: Use [Riemann sum](/integrations/integration/) over the "Input Power" sensor .
- For "Battery systems" use the "Battery Total Charge" and "Battery Total Discharge" sensors
```
